WordPress/src/main/java/org/wordpress/android/ui/posts/PostConflictResolutionFeatureUtils.kt

```diff
@@ -21,9 +25,42 @@ class PostConflictResolutionFeatureUtils @Inject constructor(
      * the remote version.
      */
     fun getRemotePostPayloadForPush(payload: RemotePostPayload): RemotePostPayload {
-        if (isPostConflictResolutionEnabled().not()) {
+        if (isPostConflictResolutionEnabled()) {
+            setLastModifiedForConflictResolution(payload)
+        } else {
             payload.shouldSkipConflictResolutionCheck = true
         }
         return payload
     }
+
+    /**
+     * Resolves post-conflict issues for scheduled posts by adjusting the `if_not_modified_since` field. Normally, when
+     * a post is scheduled for the future, both the `dateCreated` and `lastModified` fields are set to the future
+     * publication date, as returned by the backend. This setup can prevent effective post-conflict resolution.
+     * This function introduces an additional field in the payload that is used to override the `lastModified` value.
+     * By setting a past date in `if_not_modified_since`, it ensures that the conflict resolution mechanism can operate
+     * correctly.
+     */
+    private fun setLastModifiedForConflictResolution(payload: RemotePostPayload) {
+        payload.post?.let { post ->
+            payload.lastModifiedForConflictResolution = if (shouldUpdateLastModifiedForConflictResolution(post)) {
+                dateTimeUtilsWrapper.dateStringFromIso8601MinusMillis(
+                    dateTimeUtilsWrapper.currentTimeInIso8601(),
+                    MILLISECONDS_IN_A_HOUR
+                )
+            } else {
+                null
+            }
+        }
+    }
+
+    private fun shouldUpdateLastModifiedForConflictResolution(post: PostModel): Boolean {
+        return post.status == PostStatus.SCHEDULED.toString()
+                && post.remotePostId > 0
+                && post.lastModified == post.dateCreated
+    }
+
+    companion object {
+        const val MILLISECONDS_IN_A_HOUR = 1000*60*60L
```

Comment:
@zwarm I went with a hour just to be sure. If you think we should another value please feel free to suggest it :-)